server/src/test/java/org/elasticsearch/common/blobstore/fs/FsBlobContainerTests.java

```diff
@@ -95,6 +96,54 @@ public void testReadBlobRangeCorrectlySkipBytes() throws IOException {
         }
     }
 
+    public void testReadAfterBlobLengthThrowsRequestedRangeNotSatisfiedException() throws IOException {
+        final var blobName = "blob";
+        final byte[] blobData = randomByteArrayOfLength(randomIntBetween(1, frequently() ? 512 : 1 << 20)); // rarely up to 1mb
+
+        final Path path = PathUtils.get(createTempDir().toString());
+        Files.write(path.resolve(blobName), blobData);
+
+        final FsBlobContainer container = new FsBlobContainer(
+            new FsBlobStore(randomIntBetween(1, 8) * 1024, path, true),
+            BlobPath.EMPTY,
+            path
+        );
+
+        {
+            long position = blobData.length;
+            long length = randomIntBetween(1, 100);
+            try (var stream = container.readBlob(randomPurpose(), blobName, position, length)) {
+                assertThat(totalBytesRead.get(), equalTo(0L));
+                assertThat(Streams.consumeFully(stream), equalTo(0L));
+                assertThat(totalBytesRead.get(), equalTo(0L));
+            }
+        }
+
+        {
+            final long position = blobData.length + randomIntBetween(1, 100);
+            final long length = randomIntBetween(1, 100);
+            var exception = expectThrows(
+                RequestedRangeNotSatisfiedException.class,
+                () -> container.readBlob(randomPurpose(), blobName, position, length)
+            );
+            assertThat(
+                exception.getMessage(),
+                equalTo("Requested range [position=" + position + ", length=" + length + "] cannot be satisfied for [" + blobName + ']')
```

Comment:
While in S3 tests, I see a similar assertion with length-1, here I see it with just length. Why the difference?

Reply:
Because I messed up both; see latest commit that are now good.